client/extensions/hello-dolly/state/reducer.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Internal dependencies
+ */
+import {
+	ROUTE_SET,
+	SECTION_SET,
+	SITE_SETTINGS_SAVE,
+} from 'state/action-types.js';
+
+export default function( state = 0, action ) {
+	switch ( action.type ) {
+		case ROUTE_SET:
+		case SECTION_SET:
+		case SITE_SETTINGS_SAVE:
+			return advanceToNextLyric( state );
+		default:
+			return state;
+	}
+}
+
+function advanceToNextLyric( index ) {
+	return index + 1;
+}
```

Comment:
I think there's value in having a function here as it makes the code's intent more obvious. If we didn't have a function call here, we'd need a comment to explain why we're doing `index + 1`. Unless there's an important performance reason, I'd like to keep the function for this.